src/python/pants/python/python_setup.py

```diff
@@ -43,17 +43,14 @@ def register_options(cls, register):
             advanced=True,
             fingerprint=True,
             type=list,
-            fromfile=True,
+            member_type=file_option,
             default=[],
```

Comment:
A list seems strange - why would you have multiple constraint files combined as the default? - but I won't fight this one and instead just point it out. In particular, when the constraints file is being used in the spirit of a lock file, it makes 0 sense, it only makes partial sense when used with ranged constraint files that all account for different slices of the resolvable universe. That use case sounds strange but maybe it exists.

Reply:
> A nice effect of this is that we can support multiple lock files.

What does that even mean? A lockfile is a full graph exactly pinned down. To have two of these only makes sense if the two graphs are disjoint.